### PR TITLE
Moving setting the optimizer to stage 5

### LINKED ISSUE #85: Move choosing an optimizer from step 6 to step 5 in the deep learning workflow (episodes 1/2)
I feel that choosing an optimizer should be part of step 5 (currently "choose a cost function and metrics"), step 6 should just be talking about running the training, deciding on how many epochs, when to stop etc.  It makes more sense to talk about optimizer's alongside loss functions than it does with running the training.  Seeing as Keras combines choosing the optimizer with the loss function in the compile statement, it would make sense to talk about these together. 

My latest pull request for episode 1 (pr #84) has it this way, but I can change it back if others don't think this is a good idea.

- [ ] Move the compile line to step 5 
- [ ] Move explanation of choosing a optimizer to step 5

### PR DESCRIPTION
**Pull request by colinsauze**
All text and code relating to setting the optimizer are now moved into stage 5 of the process (choose a loss function and optimizer) instead of stage 6 (train the model).

fixes #85

### CODE REVIEW

**Review by dsmits — APPROVED** [review]
Looks good to me
